### DISCUSSION TITLE
XY position display, and other problems

### DISCUSSION THREAD

**Discussion by DanielTheGreat** [arkypita/LaserGRBL#1633] [Q&A]
I'm familiar with G-coding and CNC machining via Mach3, but a total novice with LaserGRBL and a Chinese VoidMicro LD2429O laser engraver.  The machine connects with LaserGRBL, I can view the GRBL parameters, do a physical home (has X,Y switches) and jog in any direction.  My initial problems are these:

1. LaserGRBL's display (top-right of image display area) of X and Y position is always 0, regardless of where I jog the head (or move it with G0...).
2. With the jog step size set to 10, I can jog 29 Y-steps to position the head just before crashing into the frame.  But if I return the head to physical home position and then issue a G0Y150 command, which should position the head about halfway to the frame end, it **crashes** into the frame (i.e. past what I thought would be Y290).
3. The physical home position is 'top-left' of the frame, viewed from what I consider the 'front' of the frame.  So positive X moves right (good), but positive Y moves 'downwards' (bad).  How do I arrange that, after issuing the $H (home) command, the head subsequently (i.e. after finding the physical home position) moves to the 'bottom-left', and the Y-axis is then inverted, so positive Y now moves 'upwards'?

Any help much appreciated!

Daniel (Australia)

**Comment by gmmanonymus111** [comment]
Could you post your `$$` GRBL settings?
For `1.`, I suspect status reporting might be incorrect
`2.`, I'm not really sure, might be lost steps due to incorrect speed/acceleration setting, or too loose/too tight timing belt
`3.`, You need to play with homing and movement masks here. Depending on your config, you need to tweak `$3` and `$23`

**Reply by DanielTheGreat** [reply]
Alex,
Thanks for your suggestions.  But arkypita has already solved the position non-display problem with a new version of LaserGRBL that handles the non-standard message format of my cheap Chinese engraver.  And others (elsewhere) have helped me (a) fix the axis directions and (b) set up a custom homing button that correctly moves the head to bottom-left after 'bouncing' off the home switches.
Daniel

**Reply by gmmanonymus111** [reply]
All right, I'm glad it worked out
